pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -333,6 +333,18 @@ public String getSchemaName() {
             + "/" + getLocalName();
     }
 
+    /**
+     * Get the full name for the topic.
+     * E.g. persistent://public/default/topic1
+     */
+    public String getFullName() {
+        if (isV2()) {
+            return String.format("%s://%s/%s/%s", domain, tenant, namespacePortion, getEncodedLocalName());
+        } else {
+            return String.format("%s://%s/%s/%s/%s", domain, tenant, cluster, namespacePortion, getEncodedLocalName());
+        }
+    }
```

Comment:
@ivankelly  To have a more consistent behavior across the codebase, I would suggest to have the "full-name" always returned by the `toString()` method.

Currently we use the `completeTopicName`, but that was the string that we got and parsed. 

Since the `TopicName` instances are cached automatically, I think we should initialize `completeTopicName` with `getFullName()` which could be a private method.